apps/api/src/services/agent-manager.service.ts

```diff
@@ -958,11 +986,28 @@ export class AgentManager {
    * @param sanitizedAgent The sanitized agent object
    * @returns The agent object with metrics attached
    */
-  attachAgentMetrics(
+  async attachAgentMetrics(
     sanitizedAgent: ReturnType<AgentManager["sanitizeAgent"]>,
   ) {
     const metadata = sanitizedAgent.metadata as AgentMetadata;
-    const stats = metadata?.stats || {};
+    const completedCompetitions = await countAgentCompetitionsForStatus(
```

Comment:
the `completedCompetitions`, `totalVotes`, `totalTrades`, and `bestPlacement` have all been added to the response. `bestPlacement` will always be undefined until the new logic mentioned above is implemented.

the follow up issue for `rank` and `score` should address populating the remaining values.